scrapinghub/client/items.py

```diff
@@ -59,3 +89,41 @@ def _modify_iter_params(self, params):
         if offset:
             params['start'] = '{}/{}'.format(self.key, offset)
         return params
+
+    def list_iter(self, chunksize=1000, *args, **kwargs):
+        """An alternative interface for reading items by returning them
+        as a generator which yields lists of items sized as `chunksize`.
+
+        This is a convenient method for cases when processing a large amount of
+        items from a job isn't ideal in one go due to the large memory needed.
+        Instead, this allows you to process it chunk by chunk.
+
+        You can improve I/O overheads by increasing the chunk value but that
+        would also increase the memory consumption.
+
+        :param chunksize: size of list to be returned per iteration
+        :param start: offset to specify the start of the item iteration
+        :param count: overall number of items to be returned, which is broken
+            down by `chunksize`.
+
+        :return: an iterator over items, yielding lists of items.
+        :rtype: :class:`collections.Iterable`
+        """
+
+        start = kwargs.pop("start", 0)
+        count = kwargs.pop("count", sys.maxsize)
+        processed = 0
+
+        while True:
+            next_key = self.key + "/" + str(start)
+            items = [
+                item for item in self.iter(
+                    count=chunksize, start=next_key, *args, **kwargs)
+            ]
+            yield items
```

Comment:
Just as a side note, when the total amount of items >= `count` and `count` isn't divisible by `chunksize`, with the current logic the function will return amount of items >= `count`. Say, `list_iter(chunksize=2, count=3)` will return 2 batches by 2 items, so 4 items in total, not 3. This is probably fine, but if keeping it as-is, I'd ask to at least mention the behaviour in the function docstring.

Reply:
Thanks @vshlapakov! Turns out there was a logic that @manycoding suggested before that wasn't added to this PR. This has been fixed and I've added a test case for it.